__tests__/app.test.js

```diff
@@ -80,4 +80,79 @@ describe('All wrong paths', () => {
             expect(response.body.message).toBe('Path not found')
         })
     })
+})
+
+describe('POST /api/articles/:article_id/comments', () => {
+    test('should return 201 status code and return the new posted comment', () => {
+        const newComment = {
+                            username: 'butter_bridge',
+                            body: 'great article'
+                            }
+        return request(app)
+        .post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(201)
+        .then((res) => {
+            expect(res.body.comment).toMatchObject({
+                                                        comment_id: 19,
+                                                        body: 'great article',
+                                                        article_id: 1,
+                                                        author: 'butter_bridge',
+                                                        votes: 0,
+                                                        created_at: expect.any(String)
+                                                    })
+        })
+    });
+    test('should return 400 Bad Request if given an article_id that does not exist',()=>{
+        const newComment = {
+            username: 'butter_bridge',
+            body: 'great article'
+            }
+        return request(app)
+        .post('/api/articles/999/comments')
+        .send(newComment)
+        .expect(400)
+        .then((res) => {
+            expect(res.body.message).toBe('Bad request')
+        })
+    })
+    test('should return 400 Bad Request if given an invalid article_id',()=>{
+        const newComment = {
+            username: 'butter_bridge',
+            body: 'great article'
+            }
+        return request(app)
+        .post('/api/articles/notAnId/comments')
+        .send(newComment)
+        .expect(400)
+        .then(({body})=>{
+            expect(body.message).toBe('Invalid ID')
+        })
+    })
+    test('should return a 406 Not acceptable if the object passed is incorrectly formatted',()=>{
```

Comment:
We haven't got a lost of acceptable values to choose from here in our server, and it's still just a bad request, which should be 400 here.